solutions/src/main/java/ru/sberbank/school/task09/RouteServiceCached.java

```diff
@@ -0,0 +1,91 @@
+package ru.sberbank.school.task09;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import lombok.NonNull;
+
+import java.io.*;
+import java.time.LocalDate;
+import java.util.*;
+
+public class RouteServiceCached extends RouteService<City, Route<City>> {
+    private Kryo kryo;
+    private String filesDirectory = "C:\\Users\\Anastasia\\Desktop\\Java\\task09Ser\\";
+    private List<String> routes = new ArrayList<>();
+
+    public RouteServiceCached() {
+        kryo = new Kryo();
+        RouteKryoSerializer kryoSerializer = new RouteKryoSerializer();
+        kryo.register(Route.class, kryoSerializer);
+        kryo.setReferences(true);
+        kryo.register(City.class);
+        kryo.register(LocalDate.class);
+        kryo.register(LinkedList.class);
+        kryo.register(ArrayList.class);
+        kryo.register(HashMap.class);
+    }
+
+    @Override
+    public Route<City> getRoute(@NonNull String from, @NonNull String to) {
+        String key = from + "_" + to;
+        Route route = null;
+        if (routes.contains(key)) {
+            route = loadRoute(key);
+        } else {
+            routes.add(key);
+            route = saveRoute(key, from, to);
+        }
+        return route;
+    }
+
+    private Route<City> loadRoute(String key) {
+        try (FileInputStream fis = new FileInputStream(filesDirectory + key + ".txt");
+             Input in = new Input(fis)) {
+            return (Route<City>) kryo.readObject(in, Route.class);
+        } catch (FileNotFoundException ex) {
+            ex.printStackTrace();
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        }
+        return null;
+    }
+
+    private Route<City> saveRoute(String key, String from, String to) {
+        Route route = super.getRouteInner(from, to);
+        try (FileOutputStream fos = new FileOutputStream(filesDirectory + key + ".txt");
+             Output out = new Output(fos)) {
```

Comment:
&#39;try resource&#39; has incorrect indentation level 13, expected level should be 16.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-908409860>*